collections/sort_by.ts

```diff
@@ -31,28 +31,35 @@ export function sortBy<T>(
     | ((el: T) => bigint)
     | ((el: T) => Date),
 ): T[] {
-  return Array.from(array).sort((a, b) => {
-    const selectedA = selector(a);
-    const selectedB = selector(b);
+  const result = array.map((value) => ({ value, selected: selector(value) }))
+    .sort((a, b) => {
```

Comment:
We can also pass the index to the selector with the following changes:
```suggestion
  const result = array.map((value, i) => ({ value, selected: selector(value, i) }))
    .sort((a, b) => {
```

Use cases are like array_multisort () in php.
https://www.php.net/manual/ja/function.array-multisort.php

```js
const targetArray = [...]
const compareArray = [...]
sortBy(targetArray, (_val, i)=>compareArray[i])
```